File: shopapp-backend/shopapp/src/main/java/com/project/shopapp/response/LoginResponse.java
package com.project.shopapp.response;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.project.shopapp.model.Role;
import com.project.shopapp.model.User;
import jakarta.persistence.Column;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class LoginResponse {

    private long id;

    @JsonProperty("message")
    private String message;

    @JsonProperty("token")
    private String token;

    @JsonProperty("token_type")
    private String tokenType = "Bearer";

    @JsonProperty("refresh_token")
    private String refreshToken;

    @JsonProperty("roles")
    private List<String> roles;

    @JsonProperty("username")
    private String username;
}

File: angular-frontend/src/app/responses/user/login.response.ts
// export interface LoginResponse {
//     message: string,
//     token: string
// }
export interface LoginResponse<T = any> {
    status: string; // "success" hoặc "error"
    message: string; // Thông báo từ server
    data: T; // Dữ liệu trả về từ server (token, user info, ...)
}